src/main.rs

```diff
@@ -43,6 +44,19 @@ fn main() -> anyhow::Result<()> {
         }
     };
 
+    if let Ok((soft_limit, hard_limit)) = getrlimit(Resource::RLIMIT_NOFILE) {
```

Comment:
Thanks! can we put this into a split out function, guard behind `#[cfg(unix)]` and call it under the `proxy` function?

Reply:
And add a comment about why we do this